src/shared/modules/connections/connectionsDuck.ts

```diff
@@ -193,7 +193,7 @@ export function getConnectionData(
     return { ...data, username: memoryUsername, password: memoryPassword }
   }
 
-  return data
+  return { ...data, username: data.username ? data.username : memoryUsername }
```

Comment:
```suggestion
  return { ...data, username: data.username ?? memoryUsername }
```

Reply:
could be written like this too I think

Reply:
Unfortunately, there are cases where `data.username` is an empty string. So, it won't be working as expected as `??` checks the value being undefined or null.